api/src/main/java/org/pmiops/workbench/monitoring/attachments/Attachment.java

```diff
@@ -0,0 +1,63 @@
+package org.pmiops.workbench.monitoring.attachments;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Collections;
+import java.util.Set;
+import org.pmiops.workbench.db.model.DbBillingProjectBufferEntry.BufferEntryStatus;
+import org.pmiops.workbench.model.DataAccessLevel;
+import org.pmiops.workbench.monitoring.views.GaugeMetric;
+import org.pmiops.workbench.monitoring.views.Metric;
+import org.pmiops.workbench.utils.Booleans;
+import org.pmiops.workbench.utils.Enums;
+
+public enum Attachment implements AttachmentBase {
+  BUFFER_ENTRY_STATUS(
+      "buffer_entry_status",
+      ImmutableSet.of(GaugeMetric.BILLING_BUFFER_PROJECT_COUNT),
+      Enums.getValueStrings(BufferEntryStatus.class)),
+  DEBUG_CONTINUOUS_VALUE("debug_continuous_value"),
+  DATASET_INVALID(
+      "dataset_invalid", ImmutableSet.of(GaugeMetric.DATASET_COUNT), Booleans.getValueStrings()),
+  USER_BYPASSED_BETA(
+      "user_bypassed_beta", ImmutableSet.of(GaugeMetric.USER_COUNT), Booleans.getValueStrings()),
+  USER_DISABLED(
+      "user_disabled", Collections.singleton(GaugeMetric.USER_COUNT), Booleans.getValueStrings()),
+  DATA_ACCESS_LEVEL(
+      "user_data_access_level",
+      ImmutableSet.of(GaugeMetric.USER_COUNT, GaugeMetric.WORKSPACE_COUNT),
+      Enums.getValueStrings(DataAccessLevel.class)),
+  WORKSPACE_ACTIVE_STATUS("workspace_active_status");
+
+  private String keyName;
+  private Set<Metric> allowedMetrics;
```

Comment:
After thinking about this some more, I don't like having the attachments know about metrics. It seems the list should be the other way around, with Metrics having a list of allowed Attachments, and Attachments only knowing about valid values.

The setup here isn't technically circular, but I don't want both enums to be "smart". So smart metrics with dumb attachments make more sense (particularly when most MeasurementBundles only have a single Metric).